src/main/cpp/t07_ip.cpp

```diff
@@ -34,5 +34,56 @@
 using namespace std;
 
 int t07_ip() {
+	int p1, p2, p3, p4;
+    int cou = 0;
+    int pos = 0;
+    int arr[4];
+    for(int i = 0; i < 4; i++)
+        arr[i] = -1;
 
+    p1 = p2 = p3 = p4 = -1;
+    string s;
+    getline(cin, s);
+
+    while((int)s.find(".", pos) > -1){
+        int pos2;
+        pos2 = s.find(".", pos);
+
+       try{
+        	arr[cou] = stoi(s.substr(pos, pos2 - pos));
+       }catch(exception e){
+            cout << "NO";
+            return 0;
+       }
+
+        cou++;
+        pos = pos2 + 1;
+    }
+    
+    if(arr[3] != -1){
+        cout << "NO";
+        return 0;
+    }
+
+    try{
+    	arr[3] = stoi(s.substr(pos));
+    }catch(exception e){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Exception should be caught by reference.](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4513044)